packages/api/src/platforms/vtex/loaders/salesChannel.ts

```diff
@@ -0,0 +1,15 @@
+import DataLoader from "dataloader";
+
+import { Options } from "..";
+import { Clients } from "../clients";
+
+import type { SalesChannel } from "./../clients/commerce/types/SalesChannel";
+
+export const getSalesChannelLoader = (_: Options, clients: Clients) => {
+  const loader = async (channels: readonly string[]) =>
+    Promise.all(
```

Comment:
`Promise.all` is all or nothing. I wonder if [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) fits better here.

Reply:
What I don't like about allSettled is that it changes the response type like in the image below. This makes me have to change all resolver's code to account for this case or throw an error on the loader, which is the same behavior of Promise.all. What do you think?

<img width="1273" alt="image" src="https://user-images.githubusercontent.com/1753396/177772669-becf00f4-6e00-40aa-9d05-0efa3dcc8721.png">